File: src/Net/Store/Before/Manager.cs
namespace Before
{
    public class Manager : Employee
    {
        public Manager(string firstName, string lastName, decimal fixedSalary)
            : base(firstName, lastName, fixedSalary)
        {
        }
        
        public decimal CalculateSalaryAfterAdditionsAndDeductions()
        {
            decimal addicionalBenefits = CalculateAddicionalBenefits();
            decimal pensionFounds = this.FixedSalary * 10 / 100;
            decimal tax = 0;
            if (FixedSalary > 3500)
                tax = FixedSalary * 5 / 100;
            return addicionalBenefits + FixedSalary - pensionFounds - tax;
        }

        private decimal CalculateAddicionalBenefits()
        {
            return this.subordinates.Count * 20;
        }
    }
}

File: src/Java/1. First Course/4. Last Hole/src/main/refactoringgolf/store/Manager.java
package refactoringgolf.store;

public class Manager extends Employee {
	
	public Manager(String firstName, String lastName, float fixedSalary)
	{
		super(firstName, lastName, fixedSalary);
	}

	public float salaryAfterAdditionsAndDeductions() {
		float addicionalBenefits = addicionalBenefits();
		float pensionFounds = this.fixedSalary * 10 / 100;
		float tax = 0;
		if (fixedSalary > 3500)
			tax = fixedSalary * 5 / 100;
		return addicionalBenefits + fixedSalary - pensionFounds - tax;
	}

	private float addicionalBenefits() {
		return this.subordinates.size() * 20;
	}
}
